libs/core/src/components/sage-switch/sage-switch.tsx

```diff
@@ -0,0 +1,140 @@
+import { Component, Element, Host, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'sage-switch',
+  styleUrl: 'sage-switch.scss',
+  shadow: true,
+})
+export class SageSwitch {
+  @Element() el: HTMLSageSwitchElement;
+
+  /**
+   * A unique identifier for this input. Associates the input with its label
+   */
+  @Prop() componentId!: string;
+
+  /**
+   * Sets state of input control
+   */
+  @Prop() checked = false;
+
+  /**
+   * Disables input control
+   */
+  @Prop() disabled? = false;
+
+  /**
+   * Text displayed with invalid state
+   */
+  @Prop() errorMessage?: string;
+
+  /**
+   * Text used for additional control description
+   */
+  @Prop() helperMessage: string;
+
+  /**
+   * Indicates when user input is not valid
+   */
+  @Prop() invalid? = false;
+
+  /**
+   * Text to be displayed as the switch label
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used for identifying form data. Unifies a group of radio inputs for toggling a single property/value
+   */
+  @Prop() name: string;
+
+  /**
+   * Forces the input to be required
+   */
+  @Prop() required? = false;
+
+  /**
+   * Attribute sent in form data. Mainly used to distinguish radio inputs
+   */
+  @Prop() value: string;
+
+  /**
+   * Specifies the underlying input element type
+   * @defaultValue 'checkbox'
+   */
+  @Prop() type: 'checkbox' | 'radio' = 'checkbox';
+
+
+  /**
+   * Generate switch classes
+   */
+  private switchClassNames = () => {
+    let switchClasses = `sage-switch`;
+
+    if (this.invalid === true) {
+      switchClasses += " sage-switch--error";
+    }
+    if (this.helperMessage !== undefined) {
+      switchClasses += " sage-switch--message";
+    }
+    return switchClasses;
+  };
+
+  /**
+   * Create id for messaging
+   */
+  private messageId = (id: string, messageType: string) => {
+    return id + `__${messageType}-message`;
+  };
+
+  /**
+   * Assign aria-description id to relate messages with input
+   */
+  private assignDescription = () => {
+    let relatedId = this.messageId(this.componentId, 'helper')
+
+    if (!this.invalid || !this.helperMessage) return;
+    if (this.invalid) relatedId = this.messageId(this.componentId, 'error');
+
+    return relatedId;
+  };
+
+  render() {
+    return (
+      <Host class={this.switchClassNames()} aria-disabled={this.disabled ? 'true' : null}>
+        <input
+          aria-describedby={this.assignDescription()}
+          aria-invalid={this.invalid ? "true" : undefined}
+          checked={this.checked}
+          class="sage-switch__input"
+          disabled={this.disabled}
+          id={this.componentId}
+          name={this.name ? this.name : this.componentId}
+          required={this.required}
+          type={this.type}
+          value={this.value}
+        />
```

Comment:
Should we consider dispatching an event here? See the checkbox or input for examples. Textarea (in progress) also has an example.

Reply:
Also just an FYI, I believe there is a validity util within the textarea as well that may be in an event here once it's ready. No action needed.

Reply:
Good call. I'd considered that, but suggest we establish a pattern for how/when events are dispatched and consumed before adding anything individually. Added an item for the consistency meeting.